src/borg/testsuite/upgrader.py

```diff
@@ -149,8 +149,8 @@ def test_convert_all(attic_repo, attic_key_file, inplace):
     """
     repo_path = attic_repo
 
-    # check should fail because of magic number
-    assert not repo_valid(repo_path)
+    with pytest.raises(Repository.AtticRepository):
+        repo_valid(repo_path)
```

Comment:
ok, i've looked it up now.
it is a bit of strange now as it can raise AtticRepository or return True/False for ok / not ok.
but i guess that is ok to deal with this special case, so keep it.